common/src/main/java/io/netty/util/Recycler.java

```diff
@@ -349,6 +358,20 @@ boolean transfer(Stack<?> dst) {
                 return false;
             }
         }
+
+        @Override
+        protected void finalize() throws Throwable {
+            super.finalize();
```

Comment:
agreed but best not to make assumptions if we can avoid it ... new finalizer makes me :cry: a bit but I can't think of a better way ATM